### ISSUE TITLE
#ifdefs break when used in structs

### ISSUE THREAD

**Issue by Nemi26** [laurentlb/Shader_Minifier#467]
I would really like to use this minifier in production, but unfortunately it can't handle my code.
Sample code that breaks:

```hlsl
struct X {
#if !defined(BLA)
    int y;
#else
    float z;
#endif
};
```

**Comment by laurentlb** [comment]
Hi,

Macros and `#if` are not well supported. In general, it's hard to parse a file when it can contain preprocessing instructions anywhere.
Do you need to keep the macros in the minified code? If not, you could preprocess the file first.

For a workaround, you can consider:
```glsl
#if !defined(BLA)
struct X {
    int y;
};
#else
struct X {
    float z;
};
#endif
```

**Comment by StreetwareGames** [comment]
I resolve different shader permutations at runtime, so yes it would be nice to have them. Using different struct definitions isn't very pretty, but I'll consider it if there are no plans to expand the macro parsing logic
